ui/app/contexts/metametrics.new.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * This file is intended to be renamed to metametrics.js once the conversion is complete.
+ * MetaMetrics is our own brand, and should remain aptly named regardless of the underlying
+ * metrics system. This file implements Segment analytics tracking.
+ */
+import React, { Component, createContext, useEffect, useCallback, useMemo } from 'react'
+import { useSelector } from 'react-redux'
+import PropTypes from 'prop-types'
+import { useLocation, matchPath } from 'react-router-dom'
+import { captureException, captureMessage } from '@sentry/browser'
+
+import { pick, omit } from 'lodash'
+import {
+  getCurrentNetworkId,
+  getAccountType,
+  getNumberOfAccounts,
+  getNumberOfTokens,
+} from '../selectors/selectors'
+import { getSendToken } from '../selectors/send'
+import {
+  txDataSelector,
+} from '../selectors/confirm-transaction'
+import { getEnvironmentType, getPlatform } from '../../../app/scripts/lib/util'
+import {
+  sendCountIsTrackable,
+  segment,
+  METAMETRICS_ANONYMOUS_ID,
+} from '../helpers/utils/metametrics.util'
+import { PATH_NAME_MAP } from '../helpers/constants/routes'
+
+export const MetaMetricsContext = createContext(() => {
+  captureException(
+    Error(`MetaMetrics context function was called from a react node that is not a descendant of a MetaMetrics context provider`),
+  )
+})
+
+const PATHS_TO_CHECK = Object.keys(PATH_NAME_MAP)
+
+export function MetaMetricsProvider ({ children }) {
+  const network = useSelector(getCurrentNetworkId)
+  const metaMetricsId = useSelector((state) => state.metamask.metaMetricsId)
+  const participateInMetaMetrics = useSelector((state) => state.metamask.participateInMetaMetrics)
+  const metaMetricsSendCount = useSelector((state) => state.metamask.metaMetricsSendCount)
+  const location = useLocation()
+
+  /**
+   * Anytime the location changes, track a page change with segment.
+   * Previously we would manually track changes to history and keep a
+   * reference to the previous url, but with page tracking we can see
+   * which page the user is on and their navigation path.
+   */
+  useEffect(() => {
+    if (location.pathname.startsWith('/initialize') || participateInMetaMetrics) {
+      // Events that happen during initialization before the user opts into MetaMetrics will be anonymous
+      const idTrait = metaMetricsId ? 'userId' : 'anonymousId'
+      const idValue = metaMetricsId ?? METAMETRICS_ANONYMOUS_ID
+      const match = matchPath(location.pathname, { path: PATHS_TO_CHECK, exact: true, strict: true })
+      if (match) {
+        const { path, params, url } = match
+        const name = PATH_NAME_MAP[path]
+        segment.page({
+          [idTrait]: idValue,
+          name,
+          properties: {
+            url,
```

Comment:
The `url` here would also contain the same private information being excluded below (the user's address)

Reply:
changed to use 'path' which will have `:address` instead of `0x0000000000000000` -- for those URLs with params objects without sensitive data in them we already get the valuable pieces.